ast/counter.go

```diff
@@ -126,6 +126,6 @@ func (ctrs counters) termCountChildren(term parser.Term, parent counter) {
 	case parser.ExtRef:
 		ctrs.count(string(t), parent)
 	default:
-		panic(fmt.Errorf("unexpected term type: %v %[1]T", t))
+		panic(fmt.Errorf("unexpected term type for counter: %v %[1]T", t))
```

Comment:
I like to prefix panics with the method name. The stack trace normally tells you anyway, but occasionally the stack doesn't appear (after calling `recover()`).
```suggestion
		panic(fmt.Errorf("counters.termCountChildren: unexpected term type: %v %[1]T", t))
```
ditto below